revolv/payments/utils.py

```diff
@@ -1,6 +1,398 @@
+
+        :return:
+            Returns a number that represents initial cash balance.
+        """
+        all_donations = Payment.objects.donations(end_date=self.start_date).aggregate(Sum('amount'))['amount__sum']
+        all_reinvestments = AdminReinvestment.objects.reinvestments(end_date=self.start_date).aggregate(Sum('amount'))['amount__sum']
+        all_repayments = AdminRepayment.objects.repayments(end_date=self.start_date).aggregate(Sum('amount'))['amount__sum']
+        cash_in_adjustments = AdminAdjustment.objects.adjustments(end_date=self.start_date, cash_type='cash_in').aggregate(Sum('amount'))['amount__sum']
+        cash_out_adjustments = AdminAdjustment.objects.adjustments(end_date=self.start_date, cash_type='cash_out').aggregate(Sum('amount'))['amount__sum']
+
+        if not all_donations:
+            all_donations = 0
+        if not all_reinvestments:
+            all_reinvestments = 0
+        if not all_repayments:
+            all_repayments = 0
+        if not cash_in_adjustments:
+            cash_in_adjustments = 0
+        if not cash_out_adjustments:
+            cash_out_adjustments = 0
+
+        cash_in = (all_donations * (1 - self.payment_service_fee_rate)) + (all_repayments * (1 - self.revolv_earnings_rate)) + cash_in_adjustments
+        cash_out  = cash_out_adjustments + all_reinvestments
+        net = cash_in - cash_out
+        return net
+
+    def clean_project_name(self, name):
+        """
+        A helper function that cleans a project name to make it JQuery 
+        friendly for use in the Accounting page. This is because the project name
+        is included as part of the class name for the selector buttons.
+        The function removes all characters that are not a-z or A-Z and replaces them
+        with a 'z'.
+
+        :args:
+            name: A string to be cleaned
+
+        :return:
+            Returns a cleaned up version of name.
+        """
+        new_name = ""
+        for c in name:
+            value = ord(c)
+            if value < 65 or value > 122:
+                new_name += 'z'
+            else:
+                new_name += c
+        return new_name
+
+    def has_payments(self, project):
+        """
+        Checks whether a project has any associated donations, reinvestments, or repayments.
+
+        :args:
+            project: The project to check.
+
+        :return:
+            A boolean of whether the project has associated payment objects.
+        """
+        donations = Payment.objects.donations(project=project, start_date=self.start_date, end_date=self.end_date).count()
+        reinvestments = AdminReinvestment.objects.reinvestments(project=project, start_date=self.start_date, end_date=self.end_date).count()
+        repayments = AdminRepayment.objects.repayments(project=project, start_date=self.start_date, end_date=self.end_date).count()
+
+        return (donations + reinvestments + repayments) != 0
+
+    def move_one_month_forward(self, date):
```

Comment:
This doesn't use `self` at all, so it should really go in a separate function, not a method of this class